velox/functions/lib/Re2Functions.cpp

```diff
@@ -1926,4 +2003,51 @@ re2ExtractAllSignatures() {
   };
 }
 
+std::shared_ptr<exec::VectorFunction> makeRe2SplitAll(
+    const std::string& name,
+    const std::vector<exec::VectorFunctionArg>& inputArgs,
+    const core::QueryConfig& /*config*/) {
+  auto numArgs = inputArgs.size();
+  VELOX_USER_CHECK_EQ(
+      numArgs, 2, "{} requires 2 arguments, but got {}", name, numArgs);
+
+  VELOX_USER_CHECK(
+      inputArgs[0].type->isVarchar(),
+      "{} requires first argument of type VARCHAR, but got {}",
+      name,
+      inputArgs[0].type->toString());
+
+  VELOX_USER_CHECK(
+      inputArgs[1].type->isVarchar(),
+      "{} requires second argument of type VARCHAR, but got {}",
+      name,
+      inputArgs[1].type->toString());
+
+  BaseVector* constantPattern = inputArgs[1].constantValue.get();
+  VELOX_USER_CHECK(
+      constantPattern != nullptr && !constantPattern->isNullAt(0),
+      "{} requires second argument of constant, but got {}",
+      name,
+      inputArgs[1].type->toString());
+
+  auto pattern = constantPattern->as<ConstantVector<StringView>>()->valueAt(0);
+
+  try {
+    return std::make_shared<Re2SplitAllConstantPattern>(pattern);
```

Comment:
so looks current implement only support constant pattern, how about if no constant ?

Reply:
Actually I have drafted a PR supporting non-constant input vectors https://github.com/facebookincubator/velox/pull/8825. Perhaps we could focus on the regex support in this PR and fix the constant vector issue in a separate PR.

Reply:
@gaoyangxiaozhu Just notice non-constant vector is also supported in your PR. Please feel free to continue its support in your PR if needed. The handling of some corner cases could be referred from mine https://github.com/facebookincubator/velox/pull/8825/files#diff-ebc054da2ecc11ac3cf20d47cbc112f9c3b9cf3d5f1784b2149fb0f4e997a2d8R127-R132. Thanks.